dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,345 @@
+* A TSpend transaction shall not be allowed in a block if it does not reach enough yes votes  which is calculated as follows: (sum(votes cast) + sum(possible remaining votes)) * TVRM / TVRD
+* A TSpend transaction shall not deplete the entire treasury. The treasury balance shall not become negative.
+* A TSpend transaction shall not spend more than 150% of avg(sum(tspends), TPE).
+* A TSpend transaction shall not be allowed in a block if the Schnorr public key is not well-known (encoded in chain parameters).
+* A TSpend transaction shall not be allowed in a block if the Schnorr signature of the transaction does not match the provided well-known public key.
+
+===Treasury Account Balance===
+
+All treasury funds in or out are subject to coinbase maturity rules and thus
+the current treasury account balance is determined by adding the sum of all
+additions and subtractions from the treasury account as of coinbase maturity
+blocks ago.
+
+The treasury account balance shall remain 0 until the treasury agenda is voted
+in and the block height reaches activation height + coinbase maturity. Only at
+that point are are any additions and subtractions possible.
+
+===Mempool===
+
+In order to allow TSpend transactions into the mempool some modifications have to
+be made.
+
+* A treasurybase transaction shall not be allowed in the mempool.
+* The mempool shall not allow more than 7 TSpend concurrent transactions in the mempool at the same time.
+* A TSpend transaction shall not be allowed in the mempool prior to stake validation height.
+* A TSpend transaction shall not be allowed in the mempool if the Expiry is not exactly on a TVI+2.
+* A TSpend transaction shall not be allowed in the mempool if the PiKey in input 0 is not well known.
+* A TSpend transaction shall not be allowed in the mempool if the signature in input 0 was not signed with the PiKey in input 0.
+* A TSpend transaction shall not be allowed in the mempool if it was mined in an ancestor block.
+
+===Version 2 Block Filters Modifications===
+
+It is possible to modify V2 instead of introducing V3 because nothing changes
+from the viewpoint of the wallet and treasury opcodes are disallowed prior to
+agenda activation.
+
+* All TSpend transaction output scripts that are tagged with OP_TGEN shall be added to the filter.
+* All valid TAdd regular transaction inputs shall be added to the filter.
+* All valid TAdd stake transaction inputs shall be added to the filter.
+* TAdd stake change shall be added to the filter.
+
+
+==Compatibility==
+
+This is a hard-forking change to the Decred consensus.  This means that once
+the agenda is voted in and becomes locked in, anybody running code that fully
+validates blocks must upgrade before the activation time or they will risk
+rejecting a chain containing a transaction that is invalid under the old rules.
```

Comment:
This was hard for me to understand. I think it should be clarified by reference to the new rules, like "or they will risk rejecting a chain containing a transaction that is valid under the new rules but invalid under the old rules."